### PR TITLE
[Docs] Revise migration guide 3/3

### PR DESCRIPTION
**Pull request by peter-csala**
# Pull Request

## The issue or feature being addressed

- #1763
- #1764

## Details on the issue fix or feature implementation
- Fixed a typo in `resilience-pipeline-registry.md`
- Fixed the tip of the _migration of retry policies_
- Fixed circuit breaker V8's warning about discontinued support for classic CB  
- Get rid of the _Migrating other policies_ section because it just echoed previous statements 

- _Migrating `Polly.Context_
   - Replaced the bullet listed predefined key mapping with a tabular fashion one
   - Added a new list item about `ResiliencePropertyKey`
   - Extended the samples with the *Bulk attach* scenario
   - Introduced constants for the context keys
   - Added a tip section to the end

- _Migrating safe execution_
   - Fixed a typo
   - Introduced constants for the context keys
   - Added a note about the discontinued support for sync safe method
   - Added a tip section to the end
    
- _Migrating no-op policies_
   - Replaced bullet listed format to a tabular one
   - Replaced `T` to `TResult` for clarity

- _Migrating policy registries_
   - Reordered the interfaces to follow alphabetical order 
   - Replaced `T` to `TKey` for clarity
   - Introduced constants for the registry keys 
   - Moved the add samples before the retrieval samples
   - Added a note about no direct update support   
   - Added a tip section to the end

- _Interoperability between policies and resilience pipeline_
   - Rephrased the paragraph 

## Open question(s)
- When I deleted the _Migrating other policies_ section then I was thinking: 
   - "Shall we add here a statement about the cache policy?"
   - What do you guys think?

**Answer: Do not mention**
   
- Regarding to the `Migrating safe execution`
   - I did not found any documentation about the `state`
   - I think we should capture _somewhere_ how does it differ from the context?; when to use which?
   - Shall we create a dedicated page with samples?
   - Or is it enough to answer the above questions with two sentences?

**Answer: Extend the resilience pipelines page**

## Confirm the following

- [x]  I started this PR by branching from the head of the default branch
- [x]  I have targeted the PR to merge into the default branch
- [ ]  I have included unit tests for the issue/feature
- [x]  I have successfully run a local build

### CODE REVIEW

**Comment by martintmk** [comment]
@peter-csala  thanks again for a great content!

Here is what I think:

> "Shall we add here a statement about the cache policy?"

I would say so, it's better than not to mention anything.

> I did not found any documentation about the state I think we should capture somewhere how does it differ from the context?; when to use which? Shall we create a dedicated page with samples?

I don't think state is something we need to mention in migration guide as it is a concept new to v8.

There is already briefly mentioned here:
https://www.pollydocs.org/advanced/performance.html#use-static-lambdas

But maybe we should expand the resilience pipeline docs with new "Executing pipeline" section where we can describe state too:
https://www.pollydocs.org/pipelines/index.html

File: docs/migration-v8.md [hunk 1]
```diff
@@ -938,82 +970,110 @@ ResilienceContextPool.Shared.Return(context);
...
 ```
 <!-- endSnippet -->
 
 ### Registry in v8
 
+> [!NOTE]
+>
+> Polly V8 does not provide an explicit API to directly update a strategy in the registry.
+>
+> On the other hand it does provide a mechanism to reload pipelines.
```
Comment (martintmk) [inline docs/migration-v8.md:1040]:
```suggestion
> On the other hand it does provide a mechanism to [reload pipelines](https://www.pollydocs.org/pipelines/resilience-pipeline-registry.html#dynamic-reloads).
```

File: docs/migration-v8.md [hunk 2]
```diff
@@ -938,82 +968,109 @@ ResilienceContextPool.Shared.Return(context);
...
+registry.GetOrAddPipeline(PipelineKey, builder => builder.AddTimeout(TimeSpan.FromSeconds(10)));
 ```
 <!-- endSnippet -->
 
+> [!TIP]
+>
+> Things to remember:
+>
+> - Use `ResiliencePipelineRegistry<TResult>` to add or get a pipelines to the registry
+> - Prefer the safe methods (for example: `TryGetPipeline{<TResult>}`) over their counterpart (for example: `GetPipeline{<TResult>}`)
```
Comment (martincostello) [inline docs/migration-v8.md:1063]:
```suggestion
> - Prefer the safer methods (for example: `TryGetPipeline{<TResult>}`) over their counterpart (for example: `GetPipeline{<TResult>}`)
```

**Review by martincostello — APPROVED** [review]

**Review by martintmk — APPROVED** [review]

**Comment by peter-csala** [comment]
> > But maybe we should expand the resilience pipeline docs with new "Executing pipeline" section where we can describe state too: https://www.pollydocs.org/pipelines/index.html
> 
> I agree that would be a great place for that.

@martintmk Will you create a PR for ☝️ ?

**Comment by martintmk** [comment]
> > > But maybe we should expand the resilience pipeline docs with new "Executing pipeline" section where we can describe state too: https://www.pollydocs.org/pipelines/index.html
> > 
> > 
> > I agree that would be a great place for that.
> 
> @martintmk Will you create a PR for ☝️ ?

I won't be able this week, will try get to it next one. You can take a shot at it if you want.

**Comment by peter-csala** [comment]
> I won't be able this week, will try get to it next one. You can take a shot at it if you want.

I am on parental leave in November as my second son was born on Monday. I will have really limited presence here in the next 3-4 weeks. If you won't have time for this during November then I will give it a try in December.

**Comment by peter-csala** [comment]
> > > I won't be able this week, will try get to it next one. You can take a shot at it if you want.
> > 
> > 
> > I am on parental leave in November as my second son was born on Monday. I will have really limited presence here in the next 3-4 weeks. If you won't have time for this during November then I will give it a try in December.
> 
> Congrats Peter and enjoy your parental leave! :) I'll try get to it if possible.

@martintmk Am I right that you did not have time to file a PR regarding ☝🏻 ?

**Comment by martintmk** [comment]
> > > > I won't be able this week, will try get to it next one. You can take a shot at it if you want.
> > > 
> > > 
> > > I am on parental leave in November as my second son was born on Monday. I will have really limited presence here in the next 3-4 weeks. If you won't have time for this during November then I will give it a try in December.
> > 
> > 
> > Congrats Peter and enjoy your parental leave! :) I'll try get to it if possible.
> 
> @martintmk Am I right that you did not have time to file a PR regarding ☝🏻 ?

Sadly, not really. Haven't had time recently.

**Pull request merged**